content/en/blog/_posts/2023-03-10-image-registry-change.md

```diff
@@ -0,0 +1,190 @@
+When you run the command above, here’s what to expect when things work correctly:
+
+```
+$ kubectl run hello-world --tty --rm -i --image=registry.k8s.io/busybox:latest sh
+If you don't see a command prompt, try pressing enter.
+/ # exit
+Session ended, resume using 'kubectl attach hello-world -c hello-world -i -t' command when the pod is running
+pod "hello-world" deleted
+```
+
+
+## What kind of errors will I see if I’m impacted? 
+
+Errors may depend on what kind of container runtime you are using, and what endpoint you are routed
+to, but it should present such as `ErrImagePull`, `ImagePullBackOff`, or a container failing to be
+created with the warning `FailedCreatePodSandBox`.
+
+Below is an example error message showing a proxied deployment failing to pull due to an unknown
+certificate:
+
+```
+FailedCreatePodSandBox: Failed to create pod sandbox: rpc error: code = Unknown desc = Error response from daemon: Head “https://us-west1-docker.pkg.dev/v2/k8s-artifacts-prod/images/pause/manifests/3.8”: x509: certificate signed by unknown authority
+```
+
+## How can I find which images are using the legacy registry?
+
+**Option 1**: See the one line kubectl command in our [earlier blog
+post](https://kubernetes.io/blog/2023/02/06/k8s-gcr-io-freeze-announcement/#what-s-next):
+
+```
+kubectl get pods --all-namespaces -o jsonpath="{.items[*].spec.containers[*].image}" |\
+tr -s '[[:space:]]' '\n' |\
+sort |\
+uniq -c
+```
+
+**Option 2**: A `kubectl` [krew](https://krew.sigs.k8s.io/) plugin has been developed called
+[`community-images`](https://github.com/kubernetes-sigs/community-images#kubectl-community-images),
+that will scan and report any images using the k8s.gcr.io endpoint.
+
+If you have krew installed, you can install it with:
+
+```
+kubectl krew install community-images
+```
+
+and generate a report with:
+
+```
+kubectl community-images
+```
+
+For alternate methods of install and example output, check out the repo:
+[kubernetes-sigs/community-images](https://github.com/kubernetes-sigs/community-image).
+
+**Option 3**: If you do not have access to a cluster directly, or manage many clusters - the best
+way is to run a search over your manifests and charts for _"k8s.gcr.io"_.
+
+**Option 4**: If you wish to prevent k8s.gcr.io based images from running in your cluster, example
+policies for [Gatekeeper](https://open-policy-agent.github.io/gatekeeper-library/website/) and
+[Kyverno](https://kyverno.io/) are available in the [AWS EKS Best Practices
+repository](https://github.com/aws/aws-eks-best-practices/tree/master/policies/k8s-registry-deprecation)
+that will block them from being pulled. You can use these third-party policies with any Kubernetes
+cluster.
+
+**Option 5**: As a **LAST** possible option, you can use a [Mutating Admission
+Webhook](https://github.com/slackhq/simple-kubernetes-webhook/blob/main/dev/gen-certs.sh) to change
+the image address dynamically. This should only be considered a stopgap till your manifests have
```

Comment:
```suggestion
**Option 5**: As a **LAST** possible option, you can use a Mutating Admission
Webhook to change the the container image dynamically, at Pod creation time
```

https://github.com/slackhq/simple-kubernetes-webhook/blob/main/dev/gen-certs.sh is a shell script and doesn't seem to be the right thing to link to. Could link to https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks though.